api/v1alpha1/wasm_types.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// Wasm defines a wasm extension.
+type Wasm struct {
```

Comment:
can we autogenerate `name`, `vm_id` and `root_id` ? and leave it out of the first iteration ?

Reply:
We can generate the `name`, but the auto-generated name might not be as meaningful as the user-provided name. 

For `vm_id` and `root_id`, we might want to expose them to the API to allows users to share VM or RootContext.

Reply:
@arkodg Envoy requires `name` to be unique within a VM. Since we have decided to leave vm_id out, EG will use a separate VM for each wasm extension, the name doesn't have to be unique in EnvoyExtensionPolicy.